fog/ingest/server/src/controller.rs

```diff
@@ -379,36 +379,39 @@ where
             // if we are still needed to be active.
             if state.is_active() {
                 log::trace!(self.logger, "publish report");
-                match self.publish_report(&ingress_pubkey, &mut state) {
-                    Ok(ingress_key_status) => {
-                        // If our key is retired, and the index we want to scan is past expiry,
-                        // early return. Note, we don't even NEED to scan
-                        // when block.index == pubkey_expiry, because the
-                        // semantic of pubkey expiry is that it bounds the tombstone block, and a
-                        // transaction cannot land in its tombstone block.
-                        // But scanning the tombstone block as well may help
-                        // deal with off-by-one errors somewhere else, and doesn't really hurt.
-                        if ingress_key_status.retired
-                            && block.index > ingress_key_status.pubkey_expiry
-                        {
-                            log::warn!(self.logger, "When preparing to process block index {}, we discovered that our ingress key is expired: {:?}. Switching to idle and nuking keys.", block.index, ingress_key_status);
-                            state.set_idle();
-                            self.new_egress_key(&mut state)
-                                .expect("Failure to rotate egress key can't be recovered from");
-                            return;
+                let mut retry_seconds = 1;
```

Comment:
this retry strategy is the same as is used lower in the file

Reply:
Why not use `retry` here? And maybe elsewhere in this file?

Reply:
we could, i guess i'm kind of ambivalent about `retry`, sometimes it is kind of boilerplatey.
it will be a little messy to code this backoff strategy as a retry iterator

when doing infinite retries i feel like just using `loop` is simpler

Reply:
i woudl be okay to switch to retry everywhere just to be consistent though. i'm just not sure its making the code better here

Reply:
Makes sense when you put it that way. Maybe document that rationale? (intentionally infinite retry, so loop makes sense)